crates/dojo-physics/src/fixed_type/vec2.cairo

```diff
@@ -0,0 +1,73 @@
+use cubit::core::FixedType;
```

Comment:
this and `bvec2` can be generalized by using generics

Reply:
That makes sense to use generics to get rid of the redundant code of the structs and the 2 methods in common `new` and `splat`. 

I'm not totally sure how to do that, with two files `vec2.cairo` and `bvec2.cairo` in separate directories, and with many other methods unique to each. If we use all from glam-rs, `Vec2` will have ~60 methods, and `BVec2` about 19.

Here's an idea but not sure:
-  make new directory `generics` and in it, new file `vec2.cairo` containing 
`struct Vec2Type<T>`, 
`trait Vec2<T>` for methods `new` and `splat`,
`impl Vec2Impl<T> of Vec2<T>` for methods `new` and `splat`

- change name of existing `vec2.cairo` to `fvec2.cairo` (for fixed-point vec2):  
`type FVec2Type = Vec2Type<FixedType>`,
`trait FVec2` for methods unique to `FVec2Type`
`FVec2Impl<T> of FVec2<T>` for methods unique to `FVec2Type`

- `bvec2.cairo`:
`type BVec2Type = Vec2Type<bool>`,
`trait BVec2` for methods unique to `BVec2Type`
`BVec2Impl<T> of BVec2<T>` for methods unique to `BVec2Type`

What are your thoughts?

Reply:
I would start simple and only implement `Vec2<T>` for now. We don't need to go for feature parity with glam-rs, I prefer to get something interesting working with `Vec2` and `Vec3`. We shouldn't need a `FVec2` impl since we should be able to do `Vec2<FixedType>`.

On naming, I think cubit is a bit unusual in adding the `Type` suffix. I would prefer a clean `Vec2`.

Reply:
Got it. That all sounds good. Will do!

Reply:
It's all done and ready to push. One question though, about how to get it to work if struct and trait have same name `Vec2<T>`.

For example, it works with:
```
struct Vec2<T> {
    x: T,
    y: T
}

trait Vec2Trait<T> {
    [...]
}
```
 but I'm not sure how to get this to work
 ```
struct Vec2<T> {
    x: T,
    y: T
}

trait Vec2<T> {
    [...]
}
```

Reply:
ah yeah that is a limitation right now, adding the `Trait` suffix is necessary